src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -384,6 +387,8 @@ private void OnBatchScopeDisposed()
                         documentsToOpen,
                         (s, documents) => s.AddDocuments(documents),
                         WorkspaceChangeKind.DocumentAdded,
+                        (s, id, newPath) => s.WithDocumentFilePath(id, newPath),
+                        WorkspaceChangeKind.DocumentChanged,
```

Comment:
Should we add a workspacechangekind for rename as well?

Reply:
There's already DocumentInfoChanged for precisely that.

Reply:
I totally missed that :)